src/python/test/test_dxpy.py

```diff
@@ -1877,6 +1877,23 @@ def test_retry_after_without_header_set(self):
         self.assertTrue(50000 <= time_elapsed)
         self.assertTrue(time_elapsed <= 70000)
 
+    @unittest.skip("Example of a request that should be retried")
```

Comment:
We could try adding the number of retries attempted as an attribute of the exception object thrown. That lets us assert on that value in tests.